SMLHelper/Patchers/EnumPatcher.cs

```diff
@@ -0,0 +1,113 @@
+﻿namespace SMLHelper.V2.Patchers
+{
+    using System;
+    using System.Collections.Generic;
+    using Harmony;
+    using SMLHelper.V2.Utility;
+
+    internal class EnumPatcher
+    {
+        internal static void Patch(HarmonyInstance harmony)
+        {
+            PatchUtils.PatchClass(harmony);
+
+            Logger.Log("EnumPatcher is done.", LogLevel.Debug);
+        }
+
+        [PatchUtils.Postfix]
+        [HarmonyPatch(typeof(Enum), nameof(Enum.GetValues))]
+        private static void Postfix_GetValues(Type enumType, ref Array __result)
+        {
+            if (enumType == typeof(TechType))
+            {
+                __result = GetValues(TechTypePatcher.cacheManager, __result);
+            }
+            else if (enumType == typeof(CraftTree.Type))
+            {
+                __result = GetValues(CraftTreeTypePatcher.cacheManager, __result);
+            }
+            else if (enumType == typeof(PingType))
+            {
+                __result = GetValues(PingTypePatcher.cacheManager, __result);
+            }
```

Comment:
`EnumPatcher` should have an `internal` method which is called by other patchers that use enums. These values should not be hardcoded here IMO.

Reply:
Nevermind, I saw your conversation with zorgesho about this.